core/blockchain.go

```diff
@@ -941,6 +941,12 @@ func (bc *BlockChain) truncateAncient(head uint64) error {
 	return nil
 }
 
+// numberHash is just a container for a number and a hash, to represent a block
+type numberHash struct {
+	number uint64
+	hash   common.Hash
+}
```

Comment:
I think I would avoid declaring this tiny helper and just use 2 sliced instead. Keeps the API surface smaller.

Reply:
I didn't address this. I disagree, keeping two paralell lists which need to be in lock-step sounds more dangerous than just one, but feel free to override this if it really irks you